packages/hint-doctype/src/hint.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * @fileoverview Hint to validate if the doctype is correct
+ */
+
+import { Category } from 'hint/dist/src/lib/enums/category';
+import { HintScope } from 'hint/dist/src/lib/enums/hintscope';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, HintMetadata, FetchEnd, IAsyncHTMLElement } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class implements IHint {
+
+    public static readonly meta: HintMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: `Hint to validate if the doctype is correct`
+        },
+        id: 'doctype',
+        schema: [],
+        scope: HintScope.any
+    }
+
+    public constructor(context: HintContext) {
+
+        const doctypeRegexFactory = (flags?: string) => {
+            return new RegExp(/(<!doctype\s(html)\s*?)(>)(.+)?/, flags ? flags : 'g');
```

Comment:
it should be `/(<!doctype\s+(html)\s*?)(>)(.+)?/`

Reply:
Yes!